components/integrations/confirm_integration/confirm_integration.jsx

```diff
@@ -145,20 +122,13 @@ export default class ConfirmIntegration extends React.Component {
                         id='add_outgoing_webhook.token'
                         defaultMessage='<b>Token</b>: {token}'
                         values={{
-                            token: IntegrationStore.getOutgoingWebhook(this.props.team.id, this.state.id).token,
+                            token: this.props.outgoingHooks[this.state.id].token,
```

Comment:
- This will break if `this.props.outgoingHooks[this.state.id]` is `undefined`. I suggests to do it like so:
```
if (this.state.type === Constants.Integrations.COMMAND) {
    const outgoingHook = this.props.outgoingHooks[this.state.id];
    if (outgoingHook) {
        headerText ...
        helpText ...
        tokenText ...
    }
}
```

Would be a plus if `undefined` can be handled with error message.

- Also should be the same for commands.
- (Not necessary) but would be a plus if the same will be implemented for incoming webhook as it's doing it directly without validating that the id is really an incoming webhook. So instead of `url: window.location.origin + '/hooks/' + this.state.id`, it could be like: `url: window.location.origin + '/hooks/' + incomingHook.id`

Reply:
Good idea, added undefined check to all 4 integration entities. Still thinking on what and how display as an error message